framework/wazuh/core/engine/queries.py

```diff
@@ -0,0 +1,535 @@
+# Copyright (C) 2015, Wazuh Inc.
+# Created by Wazuh, Inc. <info@wazuh.com>.
+# This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
+
+from typing import Any, Dict, List, Type, Optional
+
+from wazuh.core.exception import WazuhError
+from wazuh.core.utils import filter_array_by_query
+
+
+class BaseQuery:
+    """Base class for response transformations."""
+
+    def apply(self, data: List[Dict[str, Any]]) -> List[Dict[str, Any]]:
+        """Apply the transformation to the given data.
+
+        Parameters
+        ----------
+        data:  List[Dict[str, Any]]
+            The data to be transformed.
+
+        Raises
+        ------
+        NotImplementedError:
+            This method must be implemented in the derived classes.
+        """
+        raise NotImplementedError
+
+    @staticmethod
+    def should_apply(params: Dict[str, Any], data: Any) -> bool:
+        """Check if the transformation can be applied to the given parameters and data.
+
+        Parameters
+        ----------
+        params: Dict[str, Any]
+            The parameters used for the transformation.
+        data: Any
+            The data to be transformed.
+
+        Returns
+        ----------
+        bool
+            True if the transformation can be applied, False otherwise.
+
+        Raises
+        ----------
+        NotImplementedError:
+            This method must be implemented in the derived classes.
+        """
+        raise NotImplementedError
+
+
+class FieldSelector(BaseQuery):
+    """Transformation to select specific fields from the data."""
+
+    def __init__(self, params: Dict[str, Any]):
+        self.select = params['select']
+
+    @staticmethod
+    def should_apply(params: Dict[str, Any], data: Any) -> bool:
+        """ Check if the transformation can be applied to the given parameters and data.
+            The transformation can be applied if the following conditions are met:
+                - 'select' key is present in the params dictionary.
+                - 'select' value is not empty.
+                - The data is of type list.
+
+
+        Parameters
+        ----------
+        params: Dict[str, Any]
+            The parameters used for the transformation.
+        data: Any
+            The data to be transformed.
+
+        Returns
+        ----------
+        bool
+            True if the transformation can be applied, False otherwise.
+        """
+        if 'select' in params and params['select'] and isinstance(data, list):
```

Comment:
Yes, that was the initial functionality of that filter, but was changed in commit 05385cee704d656cf25a29b109fd8f419bf072b2 by request so that all classes that implement the interface have the signature, which would be to accept a `List`.